src/go/rdctl/cmd/factoryReset.go

```diff
@@ -40,23 +48,264 @@ Use the --remove-kubernetes-cache=BOOLEAN flag to also remove the cached Kuberne
-		if err != nil {
-			return err
-		}
-		fmt.Println(string(result))
-		return nil
+		return map[string]func() error{
+			"darwin":  deleteDarwinData,
+			"linux":   deleteLinuxData,
+			"windows": deleteWindowsData,
+		}[runtime.GOOS]()
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(factoryResetCmd)
 	factoryResetCmd.Flags().BoolVar(&removeKubernetesCache, "remove-kubernetes-cache", false, "If specified, also removes the cached Kubernetes images.")
+	factoryResetCmd.Flags().BoolVar(&shutdownSettings.Verbose, "verbose", false, "Be verbose")
+	factoryResetCmd.Flags().BoolVar(&shutdownSettings.DontWait, "dont-wait", false, "Don't wait for shutdown to be confirmed")
+}
+
+func deleteDarwinData() error {
+	libraryPath := path.Join(os.Getenv("HOME"), "Library")
+	appHomePath := path.Join(libraryPath, "Application Support", "rancher-desktop")
+	altAppHomePath := path.Join(os.Getenv("HOME"), ".rd")
+	cachePath := path.Join(libraryPath, "Caches", "rancher-desktop")
+	configPath := path.Join(libraryPath, "Preferences", "rancher-desktop")
+	updaterPath := path.Join(libraryPath, "Application Support", "Caches", "rancher-desktop-updater")
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath == "" {
+		logsPath = path.Join(libraryPath, "Logs", "rancher-desktop")
+	}
+
+	pathList := []string{
+		appHomePath,
+		altAppHomePath,
+		configPath,
+		logsPath,
+		updaterPath,
+		path.Join(libraryPath, "Application Support", "Rancher Desktop"),
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(altAppHomePath, path.Join(os.Getenv("HOME"), ".config"), pathList)
+}
+
+func deleteLinuxData() error {
+	homeDir := os.Getenv("HOME")
+
+	dataHomePath := os.Getenv("XDG_DATA_HOME")
+	if dataHomePath == "" {
+		dataHomePath = path.Join(homeDir, ".local", "share")
+	}
+	dataHomePath = path.Join(dataHomePath, "rancher-desktop")
+
+	configHomePath := os.Getenv("XDG_CONFIG_HOME")
+	if configHomePath == "" {
+		configHomePath = path.Join(homeDir, ".config")
+	}
+	configPath := path.Join(configHomePath, "rancher-desktop")
+
+	cacheHomePath := os.Getenv("XDG_CACHE_HOME")
+	if cacheHomePath == "" {
+		cacheHomePath = path.Join(homeDir, ".cache")
+	}
+	cachePath := path.Join(cacheHomePath, "rancher-desktop")
+
+	altAppHomePath := path.Join(homeDir, ".rd")
+
+	pathList := []string{
+		altAppHomePath,
+		configPath,
+		path.Join(configHomePath, "Rancher Desktop"),
+		path.Join(homeDir, ".local", "state", "rancher-desktop"),
+	}
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath != "" {
+		pathList = append(pathList, logsPath, path.Join(dataHomePath, "lima"))
+	} else {
+		pathList = append(pathList, path.Join(dataHomePath))
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(altAppHomePath, configHomePath, pathList)
+}
+
+// XXX: Windows doesn't have symlinks, do we have to clean up any cli-plugins?
```

Comment:
I don't think we install any docker cli plugins yet on Windows. (They show up in the path, which is good enough; and that gets taken care of by the uninstaller, not during factory reset.)

Reply:
ok